crates/core/src/operations/writer.rs

```diff
@@ -386,7 +386,7 @@ impl PartitionWriter {
                 self.flush_arrow_writer().await?;
             }
         }
-        self.arrow_writer.flush();
+        self.arrow_writer.flush()?;
```

Comment:
I've also included this non-related change, because the arrow result was not being used, but it should.